src/Dependencies/Updater.php

```diff
@@ -41,63 +27,74 @@ private function updateDependency($dep)
 
         $type = explode('/', $dep)[0];
         $name = explode('/', $dep)[1];
+
+        $result = $this->prepareForUpdate();
+        if ($result->isErr()) {
+            return $result;
+        }
+
         $depArray = $this->jsonFile->getDependency($type, $name);
         if ($depArray === []) {
             return \Result\Result::err('No matching dependency in whippet.json');
         }
 
-        $this->updateHash();
-        $this->loadGitignore();
+        return $this->update([$type=>[$depArray]]);
```

Comment:
Something feels off here: if `$depArray` is actually an array then should this be `[$type=>$depArray]`? Otherwise we have an array in an array.